routes/hot_feed.go

```diff
@@ -328,26 +395,81 @@ func (fes *APIServer) UpdateHotFeedOrderedList(
+	if len(fes.blockchain.BestChain()) > (lookbackWindowBlocks + blockOffsetForTesting) {
+		relevantNodes = fes.blockchain.BestChain()[blockTipIndex-lookbackWindowBlocks-blockOffsetForTesting : blockTipIndex]
 	}
 
-	// Iterate over the blocks and track hotness scores.
+	// Iterate over the blocks and track global feed hotness scores for each post.
+	hotnessInfoMapGlobalFeed, err := fes.PopulateHotnessInfoMap(relevantNodes, utxoView, postsToMultipliers, pkidsToMultipliers, false)
+	if err != nil {
+		glog.Infof("UpdateHotFeedOrderedList: ERROR - Failed to put PopulateHotnessInfoMap for global feed: %v", err)
+		return nil
+	}
+	// Iterate over the blocks and track tag feed hotness scores for each post.
+	hotnessInfoMapTagFeed, err := fes.PopulateHotnessInfoMap(relevantNodes, utxoView, postsToMultipliers, pkidsToMultipliers, true)
+	if err != nil {
+		glog.Infof("UpdateHotFeedOrderedList: ERROR - Failed to put PopulateHotnessInfoMap for tag feed: %v", err)
+		return nil
+	}
+	// Sort the map into an ordered list and set it as the server's new HotFeedOrderedList.
+	hotFeedOrderedList := []*HotFeedEntry{}
+	for postHashKey, hotnessInfo := range hotnessInfoMapGlobalFeed {
+		postHash := postHashKey
+		hotFeedEntry := &HotFeedEntry{
+			PostHash:     &postHash,
+			PostHashHex:  hex.EncodeToString(postHash[:]),
+			HotnessScore: hotnessInfo.HotnessScore,
+		}
+		hotFeedOrderedList = append(hotFeedOrderedList, hotFeedEntry)
+	}
+	sort.Slice(hotFeedOrderedList, func(ii, jj int) bool {
+		if hotFeedOrderedList[ii].HotnessScore != hotFeedOrderedList[jj].HotnessScore {
+			return hotFeedOrderedList[ii].HotnessScore > hotFeedOrderedList[jj].HotnessScore
+		} else {
+			return hotFeedOrderedList[ii].PostHashHex > hotFeedOrderedList[jj].PostHashHex
+		}
+	})
+	fes.HotFeedOrderedList = hotFeedOrderedList
+	fes.HotFeedPostHashToTagScoreMap = hotnessInfoMapTagFeed
+
+	// Set the ordered lists for hot feed based on tags.
+	fes.SaveOrderedFeedForTags(true)
+	// Set the ordered lists for newness based on tags.
+	fes.SaveOrderedFeedForTags(false)
+
+	// Update the HotFeedBlockHeight so we don't re-evaluate this set of blocks.
+	fes.HotFeedBlockHeight = blockTip.Height
+
+	elapsed := time.Since(start)
+	glog.Infof("Successfully updated HotFeedOrderedList in %s", elapsed)
+
+	return hotnessInfoMapGlobalFeed
+}
+
+func (fes *APIServer) PopulateHotnessInfoMap(
+	relevantNodes []*lib.BlockNode,
+	utxoView *lib.UtxoView,
+	postsToMultipliers map[lib.BlockHash]float64,
+	pkidsToMultipliers map[lib.PKID]*HotFeedPKIDMultiplier,
+	isTagFeed bool,
+) (map[lib.BlockHash]*HotnessPostInfo, error) {
 	hotnessInfoMap := make(map[lib.BlockHash]*HotnessPostInfo)
-	postInteractionMap := make(map[HotFeedInteractionKey][]byte)
+	// Map of interaction key to transaction type multiplier applied.
+	postInteractionMap := make(map[HotFeedInteractionKey]uint64)
 	for blockIdx, node := range relevantNodes {
 		block, _ := lib.GetBlock(node.Hash, utxoView.Handle)
```

Comment:
since we call PopulateHotnessInfoMap multiple times, we could optimize this by passing the relevant nodes in. GetBlock is a relatively expensive operation.